web-app/js/portal/details/SubsettingPanel.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2012 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.details');
+
+Portal.details.SubsettingPanel = Ext.extend(Ext.Panel, {
+
+    constructor : function(cfg) {
+
+        this.map = cfg.map;
+        this.mapPanel = cfg.mapPanel;
+
+        this.spatialSubsetControlsPanel = new Portal.details.SpatialSubsetControlsPanel({
+            map: cfg.map,
+            hideLabel: false
+        });
+
+        this.subsetPanelAccordion = new Portal.details.SubsetPanelAccordion();
+
+        this.emptyTextPanel =  new Portal.common.EmptyCollectionStatusPanel({
+            hidden: true
+        });
+
+        var config = Ext.apply({
+            autoScroll: true,
+            title: OpenLayers.i18n('stepHeader', { stepNumber: 2, stepDescription: OpenLayers.i18n('step2Description')}),
+            headerCfg: {
+                cls : 'steps'
+            },
+
+            items: [
+                new Ext.Spacer({height: 5}),
+                this.spatialSubsetControlsPanel,
+                this.subsetPanelAccordion,
+                new Ext.Spacer({height: 20}),
+                this.emptyTextPanel
+            ]
+        }, cfg);
+
+        Portal.details.SubsettingPanel.superclass.constructor.call(this, config);
+
+        Ext.MsgBus.subscribe(PORTAL_EVENTS.SELECTED_LAYER_CHANGED, function(eventName, openlayer) {
+            this.updateSubsetPanelAccordionItem(openlayer);
+        }, this);
+
+        Ext.MsgBus.subscribe(PORTAL_EVENTS.LAYER_REMOVED, function(eventName, openlayer) {
+            this._removeFolderForLayer(openlayer);
+        }, this);
+    },
+
+    updateSubsetPanelAccordionItem: function(layer) {
+        if (layer) {
+            if (!this._folderExistsForLayer(layer) || this._folderExistsForLayer(layer) == undefined) {
```

Comment:
Well spotted because it is weird. This is a hack because of differences between tests and running in the browser.  I have re-written this bit